tsfresh/utilities/dataframe_functions.py

```diff
@@ -372,48 +375,167 @@ def _normalize_input_to_internal_representation(timeseries_container, column_id,
+            shift_from = max(shift_until - max_timeshift - 1, 0)
+
+            df_temp = x.iloc[shift_from:shift_until] if shift_until <= len(x) else None
+        else:
+            # For negative rolling, the left side of the window moves with `timeshift`
+            shift_from = max(timeshift - 1, 0)
+            shift_until = shift_from + max_timeshift + 1
+
+            df_temp = x.iloc[shift_from:shift_until]
+
+        if df_temp is None or len(df_temp) < min_timeshift + 1:
+            return
+
+        df_temp = df_temp.copy()
+
+        # and set the shift correctly
+        if column_sort and rolling_direction > 0:
+            shift_string = "timeshift=" + str(df_temp[column_sort].iloc[-1])
+        elif column_sort and rolling_direction < 0:
+            shift_string = "timeshift=" + str(df_temp[column_sort].iloc[0])
+        else:
+            shift_string = "timeshift=" + str(timeshift - 1)
+        # and now create new ones ids out of the old ones
+        df_temp["id"] = df_temp.apply(lambda row: "id=" + str(row[column_id]) + "," + str(shift_string), axis=1)
+
+        return df_temp
+
+    return [grouped_data.apply(_f)]
+
+
+def roll_time_series(df_or_dict, column_id, column_sort=None, column_kind=None,
+                     rolling_direction=1, max_timeshift=None, min_timeshift=0,
+                     chunksize=defaults.CHUNKSIZE,
+                     n_jobs=defaults.N_PROCESSES, show_warnings=defaults.SHOW_WARNINGS,
+                     disable_progressbar=defaults.DISABLE_PROGRESSBAR,
+                     distributor=None):
     """
     This method creates sub windows of the time series. It rolls the (sorted) data frames for each kind and each id
     separately in the "time" domain (which is represented by the sort order of the sort column given by `column_sort`).
 
-    For each rolling step, a new id is created by the scheme "id={id}, shift={shift}", here id is the former id of the
-    column and shift is the amount of "time" shifts.
+    For each rolling step, a new id is created by the scheme "id={id}, timeshift={shift}", here id is the former id of
+    the column and shift is the amount of "time" shifts.
+    You can think of it as having a window of fixed length (the max_timeshift) moving one step at a time over
+    your time series.
+    Each cut-out seen by the window is a new time series with a new identifier.
 
     A few remarks:
 
      * This method will create new IDs!
-     * The sign of rolling defines the direction of time rolling, a positive value means we are going back in time
-     * It is possible to shift time series of different lengths but
+     * The sign of rolling defines the direction of time rolling, a positive value means we are shifting
+       the cut-out window foreward in time. The name of each new sub time series is given by the last time point.
+       This means, the time series named `id=4,timeshift=5` with a `max_timeshift` of 3 includes the data
```

Comment:
includes the data of the times 3,2 and 1, so it can be used to forecast time point for id=4

Reply:
No :-D
That is the point. The `id` is the old `id`, the `timeshift` is the time parameter.
So it includes the times 3, 4 and 5 and can be used to predict 6.